src/txmempool.cpp

```diff
@@ -736,22 +739,22 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
         if (fDependsWait)
             waitingOnDependants.push_back(&(*it));
         else {
-            CValidationState state;
-            assert(CheckInputs(tx, state, mempoolDuplicate, false, 0, false, NULL));
+            fCheckResult = Consensus::CheckTxInputs(tx, state, mempoolDuplicate, nSpendHeight);
```

Comment:
here, either you should add

```
fCheckResult = Consensus::CheckTxInputs(tx, state, mempoolDuplicate, nSpendHeight) || tx.IsCoinbase;
```
Or better, CheckTxInputs should do it internally.